packages/core/content-releases/server/src/services/release.ts

```diff
@@ -441,43 +443,80 @@ const createReleaseService = ({ strapi }: { strapi: LoadedStrapi }) => ({
 
     /**
      * We separate publish and unpublish actions group by content type
+     * And we keep only their ids to fetch them later to get all information needed
      */
     const actions: {
       [key: UID.ContentType]: {
-        publish: ReleaseAction['entry'][];
-        unpublish: ReleaseAction['entry'][];
+        actionsToPublishIds: ReleaseAction['entry']['id'][];
+        actionsToUnpublishIds: ReleaseAction['entry']['id'][];
       };
     } = {};
     for (const action of releaseWithPopulatedActionEntries.actions) {
       const contentTypeUid = action.contentType;
 
       if (!actions[contentTypeUid]) {
         actions[contentTypeUid] = {
-          publish: [],
-          unpublish: [],
+          actionsToPublishIds: [],
+          actionsToUnpublishIds: [],
         };
       }
 
       if (action.type === 'publish') {
-        actions[contentTypeUid].publish.push(action.entry);
+        actions[contentTypeUid].actionsToPublishIds.push(action.entry.id);
       } else {
-        actions[contentTypeUid].unpublish.push(action.entry);
+        actions[contentTypeUid].actionsToUnpublishIds.push(action.entry.id);
       }
     }
 
     const entityManagerService = strapi.plugin('content-manager').service('entity-manager');
+    const populateBuilderService = strapi.plugin('content-manager').service('populate-builder');
 
     // Only publish the release if all action updates are applied successfully to their entry, otherwise leave everything as is
     await strapi.db.transaction(async () => {
       for (const contentTypeUid of Object.keys(actions)) {
-        const { publish, unpublish } = actions[contentTypeUid as UID.ContentType];
+        // @ts-expect-error - populateBuilderService should be a function but is returning service
+        const populate = await populateBuilderService(contentTypeUid)
+          .populateDeep(Infinity)
+          .build();
+
+        const { actionsToPublishIds, actionsToUnpublishIds } =
+          actions[contentTypeUid as UID.ContentType];
+
+        /**
+         * We need to get the populate entries to be able to publish without errors on components/relations/dynamicZones
+         * Considering that populate doesn't work well with morph relations we can't get the entries from the Release model
+         * So, we need to fetch them manually
+         */
+        const actionsToPublish = (await strapi.entityService.findMany(
+          contentTypeUid as UID.ContentType,
+          {
+            filters: {
+              id: {
+                $in: actionsToPublishIds,
+              },
+            },
+            populate,
+          }
+        )) as Entity[];
+
+        const actionsToUnpublish = (await strapi.entityService.findMany(
+          contentTypeUid as UID.ContentType,
```

Comment:
I guess `entriesToPublish` would be a bit more clear. Unfortunate we have to fetch all the entries :( maybe we can find a way to improve it in the future.